### ISSUE TITLE
support basic authorization

### ISSUE THREAD

**Issue by leipert** [rexxars/registry-auth-token#4]
Hey I opened a issue on pnpm (https://github.com/rstacruz/pnpm/issues/270), so the following is shamelessly copied from there:

Hey there.
We are using a private registry which just supports basic authorization. During our tests against this registry we could not download packages.

Unfortunately you do not support Basic Authorization.

If you want, I could help you implement this (as we need it). Before I dig into it and creating a PR, I would like to know what is the best way to assist you with that feature.

FYI: [relevant parts in npm](https://github.com/npm/npm/blob/master/lib/config/get-credentials-by-uri.js)

My `.npmrc` file looks like this:

```
//registry.npmjs.org/:_authToken=TOKEN

registry=https://example.com/npm/

@npm:registry=https://example.com/npm/
//example.com/npm/:_password=PASSWORD
//example.com/npm/:username=username
//example.com/npm/:email=username@example.org
//example.com/npm/:always-auth=true

@npm-internal:registry=https://example.com/npm-internal/
//example.com/npm-internal/:_password=PASSWORD
//example.com/npm-internal/:username=username
//example.com/npm-internal/:email=username@example.org
//example.com/npm-internal/:always-auth=true
```

**Comment by rexxars** [comment]
Sure. This will cause a breaking change, as we'll need to know if the returned token is basic auth credentials or a regular bearer token.

Maybe return an object,

``` js
{type: 'Bearer', token: 'blah'}
```

Thoughts?

**Comment by leipert** [comment]
sounds nice.

If you do not want to have a breaking change, we could implement an new method:

``` js
// returns {type: 'Bearer', token: 'blah'} or {type: 'Basic', token: 'blah'}
getAuthTokenObject()
```

and issue a warning if a user uses 

``` js
// still returns only the bearer tokens
getAuthToken()
```

**Comment by rexxars** [comment]
I'm not too worried about the breaking change. With such a small module, I'd rather keep it simple. Should be an easy update for anyone using it.

**Comment by leipert** [comment]
should i write the PR?

**Comment by rexxars** [comment]
Sure, that'd be great. I'm a bit busy for the time being.

**Comment by rexxars** [comment]
Implemented in #5, published as `3.0.0`.
